python/ray/serve/scripts.py

```diff
@@ -232,49 +234,170 @@ def start(
     )
 
 
+def _generate_config_from_file_or_import_path(
+    config_or_import_path: str,
+    *,
+    name: Optional[str],
+    arguments: Dict[str, str],
+    runtime_env: Optional[Dict[str, Any]],
+) -> ServeDeploySchema:
+    """Generates a deployable config schema for the passed application(s)."""
+    if pathlib.Path(config_or_import_path).is_file():
+        config_path = config_or_import_path
+        cli_logger.print(f"Deploying from config file: '{config_path}'.")
+        if len(arguments) > 0:
+            raise click.ClickException(
+                "Application arguments cannot be specified for a config file."
+            )
+
+        # TODO(edoakes): runtime_env is silently ignored -- should we enable overriding?
+        with open(config_path, "r") as config_file:
+            config_dict = yaml.safe_load(config_file)
+            config = ServeDeploySchema.parse_obj(config_dict)
+    else:
+        # TODO(edoakes): should we default to --working-dir="." for this?
+        import_path = config_or_import_path
+        cli_logger.print(f"Deploying from import path: '{import_path}'.")
+
+        if import_path.count(":") != 1:
+            raise click.ClickException(
+                "Import path must be of the form 'module.submodule:app_or_builder', "
+                f"got: '{import_path}'."
```

Comment:
We also support import paths with a dot: `a.b.c.d` == `a.b.c:d`. This will require changing the conditional `if import_path.count(":") != 1`; otherwise, `a.b.c.d` will always raise an exception.

Reply:
Oh, let me just remove this line and leave the validation to the schema one line below :D